### ISSUE TITLE
Allow ignoring fields using struct tag

### ISSUE THREAD

**Issue by flga** [go-test/deep#36]
Hey, thank you for your time, deep is very handy!

I do have a problem tho, I need to be able to ignore certain struct fields, so I cloned the repo and implemented it locally for now, would you like a PR?

It looks something like this: (briefly)
```go
func TestStructWithTags(t *testing.T) {
	type s1 struct {
		same                    int
		modified                int
		sameIgnored             int `deep:"-"`
		modifiedIgnored         int `deep:"-"`
		ExportedSame            int
		ExportedModified        int
		ExportedSameIgnored     int `deep:"-"`
		ExportedModifiedIgnored int `deep:"-"`
	}
	type s2 struct {
		s1
		same                    int
		modified                int
		sameIgnored             int `deep:"-"`
		modifiedIgnored         int `deep:"-"`
		ExportedSame            int
		ExportedModified        int
		ExportedSameIgnored     int `deep:"-"`
		ExportedModifiedIgnored int `deep:"-"`
		recurseInline           s1
		recursePtr              *s2
	}
	sa := s2{[...]} // omitted, "same" fields are equal, "modified" are multiplied by 10
	sb := s2{[...]} // omitted, "same" fields are equal, "modified" are multiplied by 10

	// with compare unexported = true
	want := []string{
		"s1.modified: 1 != 10",
		"s1.ExportedModified: 5 != 50",
		"modified: 1 != 10",
		"ExportedModified: 5 != 50",
		"recurseInline.modified: 1 != 10",
		"recurseInline.ExportedModified: 5 != 50",
		"recursePtr.modified: 1 != 10",
		"recursePtr.ExportedModified: 5 != 50",
	}
}
```
Cheers

**Comment by daniel-nichter** [comment]
Yeah, looks like a useful feature! PR would be appreciated.

**Comment by flga** [comment]
Hey Daniel, PR sent.

Cheers

**Comment by daniel-nichter** [comment]
Released as v1.0.4
